### PR TITLE
handle error and include some traits

### PR DESCRIPTION
**Pull request by malik672**
handled error gracefully
included some necessary traits fro end users

### CODE REVIEW

File: src/behaviors/deployer.rs [hunk 1]
```diff
@@ -69,7 +69,16 @@ impl Behavior<()> for Deployer {
             pool,
         );
 
-        messager.send(To::All, serde_json::to_string(&deployment_data)?).await;
```
Comment (Autoparallel) [inline src/behaviors/deployer.rs:72]: Could just `?` here to save LOC as it is effectively the same thing

File: src/main.rs [hunk 2]
```diff
@@ -12,5 +12,5 @@ async fn main() {
     let deployer = Agent::builder("deployer").with_behavior(Deployer);
 
     world.add_agent(deployer);
-    world.run().await;
+    let _ = world.run().await;
```
Comment (Autoparallel) [inline src/main.rs:15]: For clippy?

Comment (malik672) [reply src/main.rs:15]: > For clippy?

yeah

File: src/behaviors/deployer.rs (hunk 1 above)
Comment (malik672) [reply src/behaviors/deployer.rs:72]: yeah, prefer using this ? tbh, mistake on my path

File: src/main.rs (hunk 2 above)
Comment (malik672) [reply src/main.rs:15]: is this a crate?

**Comment by malik672** [comment]
@Autoparallel i don't understand some part, why do we have to manually implement debug and clone when we can just use derive in bitmath.rs

**Comment by shuhuiluo** [comment]
> @Autoparallel i don't understand some part, why do we have to manually implement debug and clone when we can just use derive in bitmath.rs

The bindings are codegen-ed by ethers-rs.

**Comment by malik672** [comment]
> > @Autoparallel i don't understand some part, why do we have to manually implement debug and clone when we can just use derive in bitmath.rs
> 
> The bindings are codegen-ed by ethers-rs.

probably why it's getting depreceated

File: src/bindings/bit_math.rs
[the remark this answers is not in the document]
Comment (malik672) [reply src/bindings/bit_math.rs:1]: @Autoparallel yep, will close this

**Pull request closed without merging**